### PR TITLE
Plucking Flowers

### PR DESCRIPTION
**Pull request by AsthaDad**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Jay's neighbour has a garden in which there are N flowers and each flower has a beauty associated with it i.e. ith flower has B[i] beauty. Jay wants to pluck exactly K flowers from his neighbour's garden such that the sum of beauties of all flowers he plucked is maximum but to not get caught he can't pluck more than one flower in a row, more formally he can't pluck two flowers if they are adjacent.
Your task is to find maximum sum of beauty of all flowers Jay can get.

### Checklist

- [X] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#4465

### CODE REVIEW

**Comment by AsthaDad** [comment]
Review this PR  @HarshCasper   @siddharth25pandey

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            }
+        }
+        return dp[k][n];
+    }
+
+
+int main(){
+    
+        int N, K;
+        cin >> N >> K;              // size of array and max sum required
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Display message in print form as what are you taking input from the user for better UI

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                dp[i][j]=max(dp[i][j],dp[i][j-1]);
+            }
+        }
+        return dp[k][n];
+    }
+
+
+int main(){
+    
+        int N, K;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Use good naming convention

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        cin >> N >> K;              // size of array and max sum required
+        vector<int> B(N);
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Use good naming convention

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        for(int i = 0; i < N; i++){
+            cin >> B[i];                 // input array
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Display message in print form as what are you taking input from the user for better UI

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        long long ans =maxBeauty(N, K, B);
+        cout << ans << endl;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Display message in print form as what are you showing output to the user for better UI

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@AsthaDad Do the minor changes, rest is fine

File: C-Plus-Plus/cp/Plucking_Flowers.cpp [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    
+}    
+    
+/*
+
+Input:
+totalFlowers = 3
+K = 2
+Beauty[] = {1, 1000000, 1}
+
```
Comment (siddharth25pandey) [inline C-Plus-Plus/cp/Plucking_Flowers.cpp]: Sample I/O is **not matching** with the execution of the Program

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Review by siddharth25pandey — APPROVED** [review]
@AsthaDad LGTM 💯 Able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @AsthaDad for her code

**Pull request merged**